lib/puppet/util/windows/taskscheduler2.rb

```diff
@@ -0,0 +1,707 @@
+    raise Error.new(_('No currently active task. ITask is NULL.')) if @pITask.nil?
+
+    # Note Exit Code 267011 is generated when the task has never been run
+    status = @pITask.LastTaskResult
+
+    status
+  end
+
+  # Returns the comment associated with the task, if any.
+  #
+  # DONE
+  def comment
+    raise Error.new(_('No currently active task. ITask is NULL.')) if definition.nil?
+
+    definition.RegistrationInfo.Description
+  end
+
+  # Sets the comment for the task.
+  #
+  # DONE
+  def comment=(comment)
+    raise Error.new(_('No currently active task. ITask is NULL.')) if definition.nil?
+    
+    definition.RegistrationInfo.Description = comment
+
+    comment
+  end
+
+  # Returns the name of the user who created the task.
+  #
+  # DONE
+  def creator
+    raise Error.new(_('No currently active task. ITask is NULL.')) if definition.nil?
+    
+    definition.RegistrationInfo.Author
+  end
+
+  # Sets the creator for the task.
+  #
+  # DONE
+  def creator=(creator)
+    raise Error.new(_('No currently active task. ITask is NULL.')) if definition.nil?
+    
+    definition.RegistrationInfo.Author = creator
+
+    creator
+  end
+
+  # Returns a Time object that indicates the next time the task will run.
+  # nil if the task has no scheduled time
+  #
+  # DONE
+  def next_run_time
+    raise Error.new(_('No currently active task. ITask is NULL.')) if @pITask.nil?
+
+    time = @pITask.NextRunTime
+
+    # The API will still return a time WAAAY in the past if there is no schedule.
+    # As this is looking forward, if the next execution is 'scheduled' in the 1900s assume
+    # this task is not actually scheduled at all
+    time = nil if time.year < 2000
+
+    time
+  end
+
+  # Returns a Time object indicating the most recent time the task ran or
+  # nil if the task has never run.
+  #
+  # DONE
+  def most_recent_run_time
+    raise Error.new(_('No currently active task. ITask is NULL.')) if @pITask.nil?
+
+    time = @pITask.LastRunTime
+    
+    # The API will still return a time WAAAY in the past if the task has not run.
+    # If the last execution is in the 1900s assume this task has not run previosuly
+    time.year < 2000 ? nil : time
+  end
+
+  def xml_definition
+    raise Error.new(_('No currently active task. ITask is NULL.')) if @pITask.nil?
+
+    @pITask.XML
+  end
+
+  # From https://msdn.microsoft.com/en-us/library/windows/desktop/aa381850(v=vs.85).aspx
+  #
+  # The format for this string is PnYnMnDTnHnMnS, where nY is the number of years, nM is the number of months,
+  # nD is the number of days, 'T' is the date/time separator, nH is the number of hours, nM is the number of minutes,
+  # and nS is the number of seconds (for example, PT5M specifies 5 minutes and P1M4DT2H5M specifies one month,
+  # four days, two hours, and five minutes)
+  def time_limit_to_hash(time_limit)
+    regex = /^P((?'year'\d+)Y)?((?'month'\d+)M)?((?'day'\d+)D)?T((?'hour'\d+)H)?((?'minute'\d+)M)?((?'second'\d+)S)?$/
```

Comment:
I believe this is ISO8601 duration format - see the standard definition at https://en.wikipedia.org/wiki/ISO_8601#Time_intervals

Ruby has *some* support for ISO8601, but not sure about durations https://ruby-doc.org/stdlib-2.1.1/libdoc/time/rdoc/Time.html#method-c-iso8601